cdi-properties/src/main/java/org/vaadin/addon/cdiproperties/ComponentConfigurator.java

```diff
@@ -183,6 +185,40 @@ boolean appliesTo(Component component) {
         }
     }
 
+    private static class CustomPropertyDescriptionKey extends CustomProperty {
+        @Inject
+        private Instance<TextBundle> textBundle;
+        @Inject
+        private Instance<Localizer> localizer;
+
+        @Override
+        void apply(Component component, Annotation propertyAnnotation) {
+            final String descriptionKey = (String) getPropertyValue(
+                    propertyAnnotation, "descriptionKey");
+            final Boolean localized = (Boolean) getPropertyValue(
+                    propertyAnnotation, "localized");
+            if (!IGNORED_STRING.equals(descriptionKey)) {
+                TextField field = (TextField) component;
+                try {
+                    field.setDescription(textBundle.get().getText(descriptionKey));
+                    if (localized) {
+                        localizer.get().addLocalizedCaption(component,
```

Comment:
This should say `localizer.get().addLocalizedDescription(...)`
Noticed that switching the locale runtime updated the component's caption instead of the description.